Kudu.Services/Deployment/PushDeploymentController.cs

```diff
@@ -611,38 +607,50 @@ private async Task LocalZipHandler(IRepository repository, DeploymentInfoBase de
             }
         }
 
-        private async Task RetryZipDownload(IRepository repository, ArtifactDeploymentInfo zipDeploymentInfo, ITracer tracer)
+        private async Task RetryLocalZipHandler(IRepository repository, ArtifactDeploymentInfo zipDeploymentInfo, ITracer tracer)
         {
             try
             {
-                _tracer.Trace($"Starting {nameof(RetryZipDownload)}");
+                tracer.Trace($"Starting {nameof(RetryLocalZipHandler)}");
                 if (_settings.RunFromLocalZip() && zipDeploymentInfo is ArtifactDeploymentInfo)
                 {
-                    //delete existing file
                     var filePath = Path.Combine(_environment.SitePackagesPath, zipDeploymentInfo.ArtifactFileName);
+
+                    // Make sure D:\home\data\SitePackages exists
+                    FileSystemHelpers.EnsureDirectory(_environment.SitePackagesPath);
+
+                    //delete existing file
                     FileSystemHelpers.DeleteFile(filePath);
 
                     //download zip file again
-                    if (!string.IsNullOrEmpty(zipDeploymentInfo.RemoteURL))
+                    using (tracer.Step("Saving request content to {0}", filePath))
                     {
-                        await WriteSitePackageZip(zipDeploymentInfo, tracer, await DeploymentHelper.GetArtifactContentFromURLAsync(zipDeploymentInfo, tracer));
-                    }
-                    else
-                    {
-                        await WriteSitePackageZip(zipDeploymentInfo, _tracer, Request.Content);
+                        if (!string.IsNullOrEmpty(zipDeploymentInfo.RemoteURL))
+                        {
+                            var content = await DeploymentHelper.GetArtifactContentFromURLAsync(zipDeploymentInfo, tracer);
+                            await content.CopyToAsync(filePath, tracer);
+                        }
+                        else
+                        {
+                            await Request.Content.CopyFromBackupAsync(filePath, tracer, _backupStream);
```

Comment:
I would do using here and set to null.   CopyFromBackupAsync routine should not dispose the object passing to it as it does not own it.